ui-frontend/packages/catalog-ui-search/src/main/webapp/component/results-visual/result-item.tsx

```diff
@@ -611,28 +623,31 @@ export const ResultItem = ({
                   </PropertyComponent>
                 )
               })}
-              {Object.keys(lazyResult.highlights)
+
+              {lazyResult.highlights
                 .filter(
-                  attr =>
-                    attr !== 'title' &&
+                  hl =>
+                    hl.attribute !== 'title' &&
                     !customDetails.find(
-                      customDetail => customDetail.label === attr
+                      customDetail => customDetail.label === hl.attribute
                     )
                 )
-                .map(extraHighlight => {
-                  const relevantHighlight =
-                    lazyResult.highlights[extraHighlight]
+                .map(highlight => {
+                  if (shownHighlights.indexOf(highlight.attribute) !== -1) {
```

Comment:
📝  This is kind of annoying but we have to de-dupe this list (something we didn't have to do before)
That's because we want all this highlights to stick around for when we need them later. Result item is a weird corner case where we only want to show one highlight to be compact.